packages/parameters/src/ssm/SSMProvider.ts

```diff
@@ -790,6 +832,25 @@ class SSMProvider extends BaseProvider {
     return undefined;
   }
 
+  protected async _set(
+    name: string,
+    options: SSMSetOptions
+  ): Promise<number | undefined> {
+    const sdkOptions: PutParameterCommandInput = {
+      Tier: options.tier ?? 'Standard',
+      Type: options.parameterType ?? 'String',
+      Overwrite: options.overwrite ?? false,
+      ...(options.kmsKeyId ? { KeyId: options.kmsKeyId } : {}),
+      ...(options.description ? { Description: options.description } : {}),
+      ...(options?.sdkOptions ?? {}),
+      Name: name,
+      Value: options.value,
+    };
+    const result = await this.client.send(new PutParameterCommand(sdkOptions));
```

Comment:
I would like to consider wrapping this operation in a `try/catch` and throw a custom error here.

For getter methods this is done at the `BaseProvider` level and in case of error we throw a [`GetParameterError`](https://github.com/daschaa/powertools-lambda-typescript-fork/blob/ffddb693e7268b5731bbc195734a232edde976fc/packages/parameters/src/errors.ts#L1-L9).

In this case, customers will call the method directly so if it throws they'll get the error from the AWS SDK directly.

I would suggest we wrap this and throw a new `SetParameterError` (to be created) with cause the error thrown by the SDK, kind of like:
```ts
try {
  const result = await this.client.send(new PutParameterCommand(sdkOptions));
} catch (error) {
  throw new SetParameterError(`Unable to set parameter with name ${name}`, { cause: error });
}
```